internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -36,13 +37,14 @@ const (
 	errorConfigUpdate              = "error updating advanced cluster configuration options (%s): %s"
 	errorConfigRead                = "error reading advanced cluster configuration options (%s): %s"
 	ErrorClusterSetting            = "error setting `%s` for MongoDB Cluster (%s): %s"
-	ErrorAdvancedConfRead          = "error reading Advanced Configuration Option form MongoDB Cluster (%s): %s"
+	ErrorAdvancedConfRead          = "error reading Advanced Configuration Option %s for MongoDB Cluster (%s): %s"
 	ErrorClusterAdvancedSetting    = "error setting `%s` for MongoDB ClusterAdvanced (%s): %s"
 	ErrorAdvancedClusterListStatus = "error awaiting MongoDB ClusterAdvanced List IDLE: %s"
 	ErrorOperationNotPermitted     = "error operation not permitted"
 	ErrorDefaultMaxTimeMinVersion  = "default_max_time_ms can not be set for mongo_db_major_version lower than 8.0"
 	ignoreLabel                    = "Infrastructure Tool"
 	DeprecationOldSchemaAction     = "Please refer to our examples, documentation, and 1.18.0 migration guide for more details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide.html.markdown"
+	V20240530                      = "(v20240530)"
```

Comment:
Why this? Why not ErrorAdvancedConfReadLegacy with the value statically coded?

Reply:
I think in order to do that the definition of "legacy" should be clearly defined. Is it v1, is it v2 20240530 or is it v2 20240805?
Since we're using multiple versions, I'd prefer to specify the version here so it's easy to debug later.

Reply:
Good point 👍

Reply:
but aren't we "leaking" the technical detail of the API version to the customer?